### PR TITLE
Redesign configuration method

### PR DESCRIPTION
**Pull request by nblumhardt**
`WriteTo.Email()` needs a refresh, after many years of incremental additions.

This PR renames and reorganizes the configuration options to bring them up to a reasonable standard.

| Parameter              | Description                                                                                                                        |
|------------------------|------------------------------------------------------------------------------------------------------------------------------------|
| `from`                 | The email address emails will be sent from.                                                                                        |
| `to`                   | The email address emails will be sent to. Multiple addresses can be separated with commas or semicolons.                           |
| `host`                 | The SMTP server to use.                                                                                                            |
| `port`                 | The port used for the SMTP connection. The default is 25.                                                                          |
| `connectionSecurity`   | Choose the security applied to the SMTP connection. This enumeration type is supplied by MailKit. The default is `Auto`.           |
| `credentials`          | The network credentials to use to authenticate with the mail server.                                                               |
| `subject`              | A message template describing the email subject. The default is `"Log Messages"`.                                                  |
| `body`                 | A message template describing the format of the email body. The default is `"{Timestamp} [{Level}] {Message}{NewLine}{Exception}"`. |
| `formatProvider`       | Supplies culture-specific formatting information. The default is to use the current culture.                                       |

### CODE REVIEW

File: src/Serilog.Sinks.Email/Serilog.Sinks.Email.csproj [hunk 1]
```diff
@@ -5,6 +5,9 @@
     <Authors>Serilog Contributors</Authors>
     <TargetFrameworks Condition=" '$(OS)' == 'Windows_NT'">net462;net471</TargetFrameworks>
     <TargetFrameworks>$(TargetFrameworks);netstandard2.1;netstandard2.0;net5.0;net6.0;net7.0</TargetFrameworks>
```
Comment (bartelink) [inline src/Serilog.Sinks.Email/Serilog.Sinks.Email.csproj:7]:
```suggestion
    <TargetFrameworks>$(TargetFrameworks);netstandard2.0;net6.0</TargetFrameworks>
```

File: src/Serilog.Sinks.Email/Serilog.Sinks.Email.csproj [hunk 2]
```diff
@@ -5,6 +5,9 @@
     <Authors>Serilog Contributors</Authors>
     <TargetFrameworks Condition=" '$(OS)' == 'Windows_NT'">net462;net471</TargetFrameworks>
```
Comment (bartelink) [inline src/Serilog.Sinks.Email/Serilog.Sinks.Email.csproj:6]: is the net471 for a reason ?

Comment (nblumhardt) [reply src/Serilog.Sinks.Email/Serilog.Sinks.Email.csproj:6]: None at all; sorting it out now, thanks :-)

**Comment by nblumhardt** [comment]
Liking the `net462;netstandard2.0;net6.0` TFM combo :+1:

**Pull request merged**